packages/web/src/new/scenes/Main/Auth/MnemonicAdapter.tsx

```diff
@@ -0,0 +1,192 @@
+// @TODO might not need this decorator
+import type { WalletName } from '@solana/wallet-adapter-base';
+import { BaseMessageSignerWalletAdapter, WalletReadyState } from '@solana/wallet-adapter-base';
+import type { Signer, Transaction } from '@solana/web3.js';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import bs58 from 'bs58';
+import { pbkdf2 } from 'crypto';
+import { singleton } from 'tsyringe';
+import nacl from 'tweetnacl';
+
+import type { ConnectConfig, StorageInfo } from 'new/scenes/Main/Auth/typings';
+import { getKeyPairFromSeed, getStorageValue, setStorageValue } from 'new/scenes/Main/Auth/utils';
+import { notImplemented } from 'new/utils/decorators';
+
+export interface Wallet {
+  signTransaction(tx: Transaction): Promise<Transaction>;
+  signAllTransactions(txs: Transaction[]): Promise<Transaction[]>;
+  publicKey: PublicKey;
+}
+
+@singleton()
+export class MnemonicAdapter extends BaseMessageSignerWalletAdapter {
+  name = 'MnemonicAdapter' as WalletName;
+  icon = '';
+  url = '';
+  private _account: Signer | null = null;
+  private _connecting = false;
+  private _readyState = WalletReadyState.NotDetected;
+
+  static walletIndex = 0;
+  static mnemonicStrength = 256;
+
+  private static _noKeypairError = 'No keypair to sign transactions';
+  private static _mnemonicStorageKey = 'encryptedSeedAndMnemonic';
+  private static _privateStorageKey = 'walletPrivateKey';
+
+  constructor() {
+    super();
+  }
+
+  static getLocalSigner(): Signer | null {
+    const secretKeyStored = getStorageValue<Uint8Array>(MnemonicAdapter._privateStorageKey);
+
+    if (secretKeyStored) {
+      const secretKey = Uint8Array.from(Object.values(secretKeyStored));
+      const keyPair = Keypair.fromSecretKey(secretKey);
+
+      return {
+        publicKey: new PublicKey(keyPair.publicKey),
+        secretKey: keyPair.secretKey,
+      };
+    }
+
+    return null;
+  }
+
+  async signTransaction(transaction: Transaction): Promise<Transaction> {
+    if (this._account) {
+      transaction.partialSign(this._account);
+
+      return Promise.resolve(transaction);
+    }
+
+    return Promise.reject(MnemonicAdapter._noKeypairError);
```

Comment:
If you mark method `async` you can just `throw` error

Reply:
same as above